### PR TITLE
Ensure Statistical Invariance to Tensor Parallelism

### PR DESCRIPTION
**Pull request by RylieWeaver**
@XiaoWang-Github 
By default, PyTorch initializes `nn.Linear()` weights and biases with a statistical distribution parametrized by the input dimension ([link](https://docs.pytorch.org/docs/stable/generated/torch.nn.Linear)). Tensor parallelism will distort this initialization by only revealing the local dimension of (`dim // tensor_par_size`).

By correctly re-initializing the parameters to account for the `tensor_par_size`, there should be more statistical stability in activations and gradients, making for better training.

I unfortunately haven't been able to test these changes since I don't use conda after the licensing changes; discussion could follow on how is easiest/best to validate the proposed changes.

Black reformatted much of the `attention.py` and `mlp.py` files and I'm not sure whether to push that - should I be using any particular version of black?

### CODE REVIEW

**Comment by XiaoWang-Github** [comment]
Thank you.  I will investigate the above issue, and then get back to you.
